scripts/common/fetch_primers.py

```diff
@@ -134,53 +133,128 @@ def _write_primer(primer: str, bed_row: dict[str, str], seq_idx: int) -> None:
+    schemes[scheme_name_version] = {
+        FASTA: scheme_fasta_path,
+        PICKLE: scheme_pickle_path,
+    }
+    return schemes
+
+
+def raise_if_not_dir(source_schemes_path: Path):
+    if not source_schemes_path.is_dir():
+        raise FileNotFoundError("Path to the source scheme was not found. Check input arguments")
+
+
+def process_sars_cov_2_epi2me_labs_primers(
+    base_path: Path, source_schemes: str, dest_schemes_path: Path
+) -> SCHEME_TYPE:
+    """
+    Process SARS-CoV-2 primer schemes in the epi2me-labs repo
     """
-    print("Copy primers:")
     schemes = {}
+    source_schemes_path = base_path / source_schemes
+    raise_if_not_dir(source_schemes_path)
+
     for source_scheme_path in source_schemes_path.iterdir():
         scheme_name = source_scheme_path.name
         for version_path in source_scheme_path.iterdir():
-            version_name = version_path.name
-            # e.g. V4.1 => V4
-            version_name = version_name.replace(".", "-")
-            dest_scheme_path = dest_schemes_path / scheme_name / SARS_COV_2 / version_name
-            dest_scheme_path.mkdir(parents=True, exist_ok=True)
-            for primer_file in version_path.iterdir():
-                shutil.copy2(primer_file, dest_scheme_path)
-
-            ref_fasta = dest_scheme_path / f"{SARS_COV_2}.reference.fasta"
-            scheme_bed = dest_scheme_path / f"{SARS_COV_2}.scheme.bed"
-            # file to be generated
-            scheme_name_version = f"{scheme_name}_{version_name}"
-            scheme_fasta_path = dest_scheme_path / f"{SARS_COV_2}.scheme.fasta"
-            scheme_pickle_path = dest_scheme_path / f"{SARS_COV_2}.scheme.pickle"
-            extract_primer_sequences(ref_fasta, scheme_bed, scheme_fasta_path, scheme_name_version)
-            # create an ahocorasick automaton object so that this is can
-            # be loaded quickly for all samples
-            automaton = create_automaton(scheme_fasta_path, "fasta")
-            # Serialise (pickle) the ahocorasick automaton object
-            dump_pickle(scheme_pickle_path, automaton)
-            schemes[scheme_name_version] = {
-                "fasta": scheme_fasta_path,
-                "pickle": scheme_pickle_path,
-            }
-
-    scheme_index_path = dest_schemes_path / f"{SARS_COV_2}_primer_index.csv"
+            schemes.update(prepare_dest_files(dest_schemes_path, scheme_name, SARS_COV_2, version_path))
+    return schemes
+
+
+def process_sars_cov_2_quick_lab_primers(base_path: Path, source_schemes: str, dest_schemes_path: Path) -> SCHEME_TYPE:
+    """
+    Process SARS-CoV-2 ARTIC primer schemes in the quick-lab repo
+    """
+    schemes = {}
+    source_schemes_path = base_path / source_schemes
+    raise_if_not_dir(source_schemes_path)
+
+    for scheme_length_dir in source_schemes_path.iterdir():
+        # these schemes are separated by length
+        if scheme_length_dir.name not in ["400", "1200"]:
```

Comment:
I'm thinking the "400" and "1200" should be configuration in the script, another number could be added in future.

Reply:
Those numbers depend on the base length as defined in the repository. I'd prefer that we hard code them and if the repository changes, we will need a PR anyway as we need to control the commit.